arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/ops/CallbackStreamTest.kt

```diff
@@ -0,0 +1,162 @@
+package arrow.fx.coroutines.stream.ops
+
+import arrow.fx.coroutines.CancelToken
+import arrow.fx.coroutines.ForkAndForget
+import arrow.fx.coroutines.Promise
+import arrow.fx.coroutines.Schedule
+import arrow.fx.coroutines.StreamSpec
+import arrow.fx.coroutines.assertThrowable
+import arrow.fx.coroutines.milliseconds
+import arrow.fx.coroutines.parTupledN
+import arrow.fx.coroutines.sleep
+import arrow.fx.coroutines.stream.Stream
+import arrow.fx.coroutines.stream.callbackStream
+import arrow.fx.coroutines.stream.cancellableCallbackStream
+import arrow.fx.coroutines.stream.chunk
+import arrow.fx.coroutines.stream.compile
+import arrow.fx.coroutines.throwable
+import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.filter
+import io.kotest.property.arbitrary.int
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.startCoroutine
+
+class CallbackStreamTest : StreamSpec(iterations = 250, spec = {
+
+  "callbackStream should be lazy" {
+    var effect = 0
+    Stream.callbackStream<Int> {
+      effect += 1
+    }
+
+    effect shouldBe 0
+  }
+
+  "emits values" {
+    Stream.callbackStream {
+      emit(1)
+      emit(2)
+      emit(3)
+    }
+      .take(3)
+      .compile()
+      .toList() shouldBe listOf(1, 2, 3)
+  }
+
+  "emits varargs" {
+    Stream.callbackStream {
+      emit(1, 2, 3)
+    }
+      .take(3)
+      .compile()
+      .toList() shouldBe listOf(1, 2, 3)
+  }
+
+  "emits iterable" {
+    Stream.callbackStream<Int> {
+      emit(listOf(1, 2, 3))
+    }
+      .take(3)
+      .compile()
+      .toList() shouldBe listOf(1, 2, 3)
+  }
+
+  "emits chunks" {
+    checkAll(Arb.chunk(Arb.int()).filter { it.isNotEmpty() }, Arb.chunk(Arb.int()).filter { it.isNotEmpty() }) { ch, ch2 ->
```

Comment:
This test shows that emitting `Chunk.empty` breaks this constructor. This should be fixed, or we should create a ticket for it if we want to unblock @JorgeCastilloPrz by merging.

This can be fixed by without using `null` or `Option`, by changing to `Queue.unbounded<Any?>`.
And using reference checking/casting + `object End`.